integration-tests/src/test/java/tachyon/master/JournalCrashIntegrationTest.java

```diff
@@ -0,0 +1,240 @@
+    mExecutorsForMasterInfo.shutdown();
+    mLocalClusterThreadGroup.interrupt();
+    // Clear system properties since the cluster is not stopped gracefully.
+    System.clearProperty(Constants.TACHYON_HOME);
+    System.clearProperty(Constants.WORKER_PORT);
+    System.clearProperty(Constants.WORKER_DATA_PORT);
+    System.clearProperty(Constants.WORKER_DATA_FOLDER);
+    System.clearProperty(Constants.WORKER_MEMORY_SIZE);
+    System.clearProperty(Constants.USER_REMOTE_READ_BUFFER_SIZE_BYTE);
+    System.clearProperty(Constants.WORKER_TO_MASTER_HEARTBEAT_INTERVAL_MS);
+    System.clearProperty(Constants.WORKER_MAX_TIERED_STORAGE_LEVEL);
+    System.clearProperty(Constants.WORKER_NETTY_WORKER_THREADS);
+    System.clearProperty(Constants.WORKER_MIN_WORKER_THREADS);
+    System.clearProperty(Constants.USE_ZOOKEEPER);
+    System.clearProperty(Constants.ZOOKEEPER_ADDRESS);
+    System.clearProperty(Constants.ZOOKEEPER_ELECTION_PATH);
+    System.clearProperty(Constants.ZOOKEEPER_LEADER_PATH);
+    System.clearProperty("fs.hdfs.impl.disable.cache");
+  }
+
+  @Before
+  public final void before() throws Exception {
+    System.setProperty("fs.hdfs.impl.disable.cache", "true");
+    mLocalClusterThreadGroup = new ThreadGroup("JournalCrashIntegrationTest Thread Group"){
+      @Override public void uncaughtException(Thread t, Throwable e) {
+        // Forcibly kill a thread may cause error. Ignore this since we want to crash.
+      }
+    };
+  }
+
+  /**
+   * Reproduce the journal and check if the state is correct.
+   */
+  private void reproduceAndCheckState(int successFiles, int successTables) throws IOException,
+      InvalidPathException, FileDoesNotExistException, TableDoesNotExistException {
+    String masterJournal = mMasterTachyonConf.get(Constants.MASTER_JOURNAL_FOLDER);
+    Journal journal = new Journal(masterJournal, "image.data", "log.data", mMasterTachyonConf);
+    MasterInfo info = new MasterInfo(new InetSocketAddress(9999), journal, mExecutorsForMasterInfo,
+        mMasterTachyonConf);
+    info.init();
+
+    Assert.assertEquals(successFiles, info.listFiles(new TachyonURI(TEST_FILE_DIR), false).size());
+    for (int f = 0; f < successFiles; f ++) {
+      Assert.assertTrue(info.getFileId(new TachyonURI(TEST_FILE_DIR + f)) != -1);
+    }
+    Assert.assertEquals(successTables,
+        info.listFiles(new TachyonURI(TEST_TABLE_DIR), false).size());
+    for (int t = 0; t < successTables; t ++) {
+      Assert.assertTrue(info.getRawTableId(new TachyonURI(TEST_TABLE_DIR + t)) != -1);
+    }
+    info.stop();
+  }
+
+  private LocalTachyonClusterMultiMaster setupMultiMasterCluster() throws IOException,
+      InterruptedException {
+    // Setup the cluster with the specified ThreadGroup.
+    final LocalTachyonClusterMultiMaster cluster = new LocalTachyonClusterMultiMaster(100,
+        TEST_MASTERS, TEST_BLOCK_SIZE);
+    Thread clusterThread = new Thread(mLocalClusterThreadGroup, new Runnable() {
```

Comment:
By default, new thread will be put into the same thread group of its parent. So cluster.start() will create threads in mLocalClusterThreadGroup.